### PR TITLE
DOCSP-41241: Laravel Sanctum

### PR DESCRIPTION
**Pull request by norareidy**
JIRA - https://jira.mongodb.org/browse/DOCSP-41241
Staging - https://preview-mongodbnorareidy.gatsbyjs.io/laravel/DOCSP-41241-stage/user-authentication/#customize-user-authentication

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/user-authentication.txt [hunk 1]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+uses Laravel's cookie-based authentication services.
+
+Run the following commands from your project root to install Laravel Sanctum,
+then publish and run its migration files:
+
+.. code-block:: bash
+
+   composer require laravel/sanctum
+   php artisan vendor:publish --provider="Laravel\Sanctum\SanctumServiceProvider"
+   php artisan migrate
```
Comment (GromNaN) [inline docs/user-authentication.txt]: This last command cannot be done before the model is updated to use the `DocumentModel` trait.

File: docs/user-authentication.txt [hunk 2]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+.. tip::
+
+   To learn more about installing and configuring Laravel Sanctum, see the
+   `Installation <https://laravel.com/docs/{+laravel-docs-version+}/sanctum#installation>`__
+   section of the Laravel Sanctum guide.
+
+To use Laravel Sanctum with {+odm-short+}, you must modify the ``PersonalAccessToken``
+model provided by Sanctum.
+
+Example
```
Comment (GromNaN) [inline docs/user-authentication.txt]: This is not an example but the required procedure.

**Review by GromNaN — CHANGES_REQUESTED** [review]
I would say in the introduction that Laravel Sanctum work by overriding the default model with a model that brings MongoDB features. Everything else works the same.

File: docs/user-authentication.txt [hunk 3]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
    :language: php
    :dedent:
 
-Enable Password Reminders 
--------------------------
+Customize User Authentication
+-----------------------------
+
+In this section, you can learn how to use the following features to
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: before this sentence, add a sentence explaining why someone might want to make a custom model

File: docs/user-authentication.txt [hunk 4]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+
+- :ref:`laravel-user-auth-sanctum`
+- :ref:`laravel-user-auth-reminders`
+
+.. _laravel-user-auth-sanctum:
+
+Laravel Sanctum
+~~~~~~~~~~~~~~~
+
+You can install the Laravel Sanctum package to provide API tokens to users
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: add a sentence explaining broadly what laravel sanctum is

File: docs/user-authentication.txt [hunk 5]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+and authenticate single-page applications. To manage API tokens, Sanctum stores
+each token in a database table and authenticates incoming HTTP requests through
```
Comment (rustagir) [inline docs/user-authentication.txt]:
```suggestion
each token in a database table and authenticates incoming HTTP requests by using
```

File: docs/user-authentication.txt [hunk 6]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+
+   composer require laravel/sanctum
+   php artisan vendor:publish --provider="Laravel\Sanctum\SanctumServiceProvider"
+   php artisan migrate
+
+.. tip::
+
+   To learn more about installing and configuring Laravel Sanctum, see the
+   `Installation <https://laravel.com/docs/{+laravel-docs-version+}/sanctum#installation>`__
+   section of the Laravel Sanctum guide.
```
Comment (rustagir) [inline docs/user-authentication.txt]:
```suggestion
   To learn more about installing and configuring Laravel Sanctum, see
   `Installation <https://laravel.com/docs/{+laravel-docs-version+}/sanctum#installation>`__
   in the Laravel Sanctum guide.
```

File: docs/user-authentication.txt [hunk 7]
```diff
@@ -97,8 +97,64 @@ manage user authentication, as shown in the following code:
...
+
+To use Laravel Sanctum with {+odm-short+}, you must modify the ``PersonalAccessToken``
+model provided by Sanctum.
```
Comment (rustagir) [inline docs/user-authentication.txt]: S: provide a little more explanation of what is required here

**Review by rustagir — CHANGES_REQUESTED** [review]
a couple of suggestions about flow

**Review by rustagir — APPROVED** [review]
lgtm

File: docs/user-authentication.txt [hunk 8]
```diff
@@ -97,8 +97,62 @@ manage user authentication, as shown in the following code:
...
+- :ref:`laravel-user-auth-reminders`
+
+.. _laravel-user-auth-sanctum:
+
+Laravel Sanctum
+~~~~~~~~~~~~~~~
+
+Laravel Sanctum is an authentication package that can manage API requests and
+single-page application authentication. To manage API requests, Sanctum issues
+API tokens that are stored in a database table and authenticates incoming HTTP
```
Comment (GromNaN) [inline docs/user-authentication.txt]: Are you sure to keep the word "table", it refers to the SQL database.
```suggestion
API tokens that are stored in the database and authenticates incoming HTTP
```

**Review by GromNaN — APPROVED** [review]
SGTM

**Pull request merged**